### PR TITLE
Update log4j configuration and simplify flink startup

### PR DESCRIPTION
**Pull request by mmadoo**
- Replace log4j configuration file with default flink log4j-console
- Start foreground and remove while loop (that took 1 CPU if log file is not found)
- Mount conf volume in a similar way between job manager and task manager

### CODE REVIEW

**Comment by s-vkropotko** [comment]
Hello @mmadoo!

Thank you for your contribution!
About changing startup scripts. It is implemented this way to support HA configuration. 
I personally don't remember why it required this tricks with `sed`

I see that HA mode doesn't work. Besides this mistake:
https://github.com/riskfocus/helm-charts-public/pull/19/files#diff-683bf8e4cfc9238bbd5089656baf6f15ba79aadd8d7a14bd8b713d6fe0e629c4R145
You also forgot to update this part:
https://github.com/riskfocus/helm-charts-public/pull/19/files#diff-5afe8dfcf149fcc9d435188385fc9cfc480791ca8a5cdddd39bfd04f4fc7a45cR150

After this updates non-HA configuration works.

But HA configuration is still failing with `Exit code 1` and logs:
```
Starting taskexecutor as a console application on host flink-taskmanager-1.
log4j:WARN No appenders could be found for logger (org.apache.flink.runtime.taskexecutor.TaskManagerRunner).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.```
Are you sure that configuration of log4j is valid?
```

If you have any point against HA configuration or you know how to configure it properly please leave your comment here or update PR with correct information (update README.md in this chart)

**Comment by mmadoo** [comment]
I do not use HA, so I am unable to test.
According to [documentation ](https://ci.apache.org/projects/flink/flink-docs-stable/ops/config.html#basic-setup), jobmanager.rpc.address is ignored in HA setup, why does the chart should set this parameter in this case ?

For log4j, I have no issue. In flink-console.sh (from image flink:1.11.1-scala_2.12), logging option is configured using:
```
log_setting=("-Dlog.file=${log}" "-Dlog4j.configuration=file:${FLINK_CONF_DIR}/log4j-console.properties" "-Dlog4j.configurationFile=file:${FLINK_CONF_DIR}/log4j-console.properties" "-Dlogback.configurationFile=file:${FLINK_CONF_D
IR}/logback-console.xml")
```
so log4j uses conf/log4j-console.properties and not conf/log4.properties.
An option, could be to mount both log4j.properties and log4j-console.properties using;
```
              - key: log4j-console.properties
                path: log4j.properties
```

**Comment by mproch** [comment]
Hi, I think the problem with logging is that Flink 1.11 uses log4j2 by default, which has different config format than old log4j, used by older Flink versions...

File: flink/templates/jobmanager.yaml [hunk 1]
```diff
@@ -110,19 +110,7 @@ spec:
             periodSeconds: {{ .Values.jobmanager.livenessProbe.periodSeconds }}
           volumeMounts:
             - name: flink-config-volume
-              mountPath: {{ .Values.flink.workDir }}/conf/flink-conf.yaml.tpl
```
Comment (s-vkropotko) [inline flink/templates/jobmanager.yaml]: I prefer to keep existed definition to keep an ability for additional configuration for flink-conf.yaml with template file.

File: flink/values.yaml [hunk 2]
```diff
@@ -111,14 +143,7 @@ jobmanager:
     # 'sed' can be changed to use flink service name, e.g. {{ include "flink.name" . }}-jobmanager
     command: >-
       sed 's/REPLACE_HOSTNAME/'${FLINK_POD_IP}'/'
-      $FLINK_HOME/conf/flink-conf.yaml.tpl > $FLINK_HOME/conf/flink-conf.yaml &&
```
Comment (s-vkropotko) [inline flink/values.yaml:134]: plz keep this line

File: flink/values.yaml [hunk 3]
```diff
@@ -150,14 +175,7 @@ jobmanager:
     wget https://repo1.maven.org/maven2/net/java/dev/jna/jna-platform/5.4.0/jna-platform-5.4.0.jar
     -O /opt/flink/lib/jna-platform-5.4.0.jar
   command: >-
-    cp $FLINK_HOME/conf/flink-conf.yaml.tpl $FLINK_HOME/conf/flink-conf.yaml &&
```
Comment (s-vkropotko) [inline flink/values.yaml:172]: plz keep this line

File: flink/values.yaml [hunk 4]
```diff
@@ -16,18 +16,50 @@ flink:
...
-      log4j.logger.org.apache.kafka.clients.producer.ProducerConfig=WARN
-      log4j.logger.org.apache.kafka.clients.consumer.ConsumerConfig=WARN
-      log4j.logger.org.apache.hadoop=INFO
-      log4j.logger.org.apache.zookeeper=INFO
-      log4j.appender.file=org.apache.log4j.FileAppender
-      log4j.appender.file.file=${log.file}
-      log4j.appender.file.layout=org.apache.log4j.PatternLayout
-      log4j.appender.file.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss,SSS} %-5p %-60c %x - %m%n
-      log4j.logger.org.apache.flink.shaded.akka.org.jboss.netty.channel.DefaultChannelPipeline=ERROR, file
+      # This affects logging for both user code and Flink
```
Comment (s-vkropotko) [inline flink/values.yaml:19]: If we provide a chart with non-default logging setup it definitely should be in README file.
So could you please update README file

**Review by s-vkropotko — CHANGES_REQUESTED** [review]
I think this update for logging (write to console + write to rotated files on filesystem(if I understand it correct)) is good decision. 
But we have some trick to work with HA configuration and I would like to save this also.
I suggest to keep existed structure and update only:
1) logging configuration
2) Name of the item in configmap
3) Name of the mounted files for Taskmanager and Jobmanager
For Jobmanager you need to change also subPath and mountPath:
https://github.com/riskfocus/helm-charts-public/blob/master/flink/templates/jobmanager.yaml#L116
4) Remove `while` loop from all commands (HA also)

**Comment by s-vkropotko** [comment]
Hi @mmadoo !
Could you please check my comments?

File: flink/values.yaml (hunk 4 above)
Comment (mmadoo) [reply flink/values.yaml:19]: Flink provides multiple logging files in binary distribution. 
This merge request uses log4j-console.properties while master is using log4j.properties.
Thus, both configuration can be seen as default logging setup...

**Comment by s-vkropotko** [comment]
Thank you @mmadoo !
Great work

**Pull request merged**